### DISCUSSION TITLE
Disable date issue in igx-date-picker

### DISCUSSION THREAD

**Discussion by kevalgw** [IgniteUI/igniteui-angular#11924] [Q&A]
I have disabled the back date in date picker. it is disabled in calendar.
But it is allowing me to enter the disabled date manually. Is there any way to control this behavior???

I have attached the ss for reference  

1. Here the date is disabled as per requirement
![image](https://user-images.githubusercontent.com/101707048/182364028-97572931-744e-4bbf-8df0-eb35e36cfa99.png)

2. But i am able to enter the disabled date manually by typing
![image](https://user-images.githubusercontent.com/101707048/182364548-fad2c75c-a145-4bf6-8ab8-99c89a610189.png)
![image](https://user-images.githubusercontent.com/101707048/182364837-f39c6fc0-ecc2-4706-8e7a-72b59c634dac.png)

3. Following is the html tag
<igx-date-picker
                  mode="dropdown"
                  [mask]="'dd-MM-yyyy'"
                  [format]="'dd-MM-yyyy'"
                  formControlName="date"
                  [disabledDates]="disabledDate" 
                  (onSelection)="function($event)"
                  (valueChange)="function($event)">
                  <label igxLabel>Dep Date</label>
</igx-date-picker>

**Comment by RivaIvanova** [comment]
Hello @kevalgw,

Using the IgxDatePicker’s **disabledDates** property gets/sets the disabled dates descriptors and prevents a date to be selected from the dropdown dialog, however, this date could be typed manually in the input field and how this behavior should be handled is up to the developer.

There are a few options depending on your requirement. For example, you could use the **readOnly** property and set it to **true**. This would disable the input field and selecting a date could be performed only from the dropdown.

Another approach is using the **[minValue](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxDatePickerComponent.html#minValue)** property along with the **[validationFailed](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxDatePickerComponent.html#validationFailed)** event which is emitted when the user types an invalid date in the date-picker editor. When this event is emitted a certain action could be performed, e.g., clearing the input field `this.datePicker.clear()` or selecting the current or another date `this.datePicker.select(date)`.

A similar behavior could be achieved by using the **[valueChange](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxDatePickerComponent.html#valueChange)** event which is emitted when the date-picker's value changes.

[Here](https://stackblitz.com/edit/angular-urpsar?file=src%2Fapp%2Fdatepicker-sample-1%2Fdatepicker-sample-1.component.ts,src%2Fapp%2Fdatepicker-sample-1%2Fdatepicker-sample-1.component.html) could be found a small sample demonstrating the above approaches. I hope it helps!

**Reply by kevalgw** [reply]
Hi @RivaIvanova 

Is [readOnly] = "true" supported in version 10??

**Reply by RivaIvanova** [reply]
Hello @kevalgw,

With every new version of Ignite UI for Angular, the developer teams aim to improve the product by providing new features and functionalities, as well as improving the currently available ones and with this in mind, if this property is not recognized in your application, then it is probably introduced in a later version.

Additionally, please keep in mind that **version 10.x is considered retired,** and I would suggest updating your application to the latest (or a newer and supported one) in order to take advantage of all new features and functionalities. For more information, you can refer to our official [Update guide](https://www.infragistics.com/products/ignite-ui-angular/angular/components/general/update-guide).

However, in case you want to stick to Ignite UI for Angular version 10, I would suggest referring to our official documentation for this version, where you could find detailed information about every component and its properties, methods, events, etc. The documentation for version 10.0.x could be found [here](https://www.infragistics.com/products/ignite-ui-angular/docs/10.0.12/typescript/).

**Reply by kevalgw** [reply]
Hello @RivaIvanova 

Thanks for the solution
